### PR TITLE
[2669] Introduce ChannelItem class

### PR DESCRIPTION
**Pull request by bychkovdmitry**
https://github.com/GetStream/stream-chat-android/issues/2669

### 🎯 Goal

Introduce `ChannelItem` class that represents an item on the channel list screen.

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [x] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [x] Affected documentation updated (docusaurus, tutorial, CMS (task created))
- [x] Reviewers added

### CODE REVIEW

File: stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ChannelActivity.kt [hunk 1]
```diff
@@ -202,6 +203,7 @@ class ChannelActivity : AppCompatActivity() {
                         .wrapContentHeight()
                         .align(Alignment.Center),
                     shape = RoundedCornerShape(16.dp),
+                    isMuted = listViewModel.isChannelMuted(selectedChannel!!.cid),
```
Comment (filbabic) [inline stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ChannelActivity.kt:206]: Let's not do this. I know it's a bit uglier, but I'd rather capture a value and then check that, than have `!!` in code.

Something like:

```
val currentSelectedChannel = selectedChannel

if (currentSelectedChannel != null) {
 // ...
}
```

The state delegate isn't the best in this case, but I think this is okay too.

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/channel/ChannelsScreen.kt [hunk 2]
```diff
@@ -141,6 +141,7 @@ public fun ChannelsScreen(
                     .align(Alignment.BottomCenter),
                 selectedChannel = selectedChannel!!,
                 currentUser = user,
+                isMuted = listViewModel.isChannelMuted(selectedChannel!!.cid),
```
Comment (filbabic) [inline stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/channel/ChannelsScreen.kt:144]: Same here. Worst case, we could use one of the capturing functions such as let, but I'd still prefer a simple capturing value.

**Review by filbabic — CHANGES_REQUESTED** [review]
Would love to get rid of the !! operator, but the rest of it

File: stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ChannelActivity.kt (hunk 1 above)
Comment (bychkovdmitry) [reply stream-chat-android-compose-sample/src/main/java/io/getstream/chat/android/compose/sample/ChannelActivity.kt]: Yes, works for me. Implemented like that at fist, but this produces IDE warning. Done, captured the value in a temp `currentSelectedChannel` value.

**Review by filbabic — APPROVED** [review]
Looks good! :]

**Pull request merged**